docs/01-command-line-reference.md

```diff
@@ -0,0 +1,247 @@
+---
+title: Command Line Interface
+---
+
+Rollup should typically be used from the command line. You can provide an  optional Rollup configuration file to simplify command line usage and enable advanced Rollup functionality.
+
+### Configuration Files
+
+Rollup configuration files are optional, but they are powerful and convenient and thus **recommended**.
+
+A config file is an ES6 module that exports a default object with the desired options. Typically, it is called `rollup.config.js` and sits in the root directory of your project.
+
+Also you can use CJS modules syntax for the config file.
+
+```javascript
+module.exports = {
+  input: 'src/main.js',
+  output: {
+    file: 'bundle.js',
+    format: 'cjs'
+  }
+};
+```
+
+It may be pertinent if you want to use the config file not only from the command line, but also from your custom scripts programmatically.
+
+Consult the [big list of options](guide/en#big-list-of-options) for details on each option you can include in your config file.
+
+```javascript
+// rollup.config.js
+
+export default { // can be an array (for multiple inputs)
+  // core input options
+  input,     // required
+  external,
+  plugins,
+
+  // advanced input options
+  onwarn,
+  perf,
+
+  // danger zone
+  acorn,
+  acornInjectPlugins,
+  treeshake,
+  context,
+  moduleContext,
+
+  // experimental
+  experimentalCodeSplitting,
+  manualChunks,
+  optimizeChunks,
+  chunkGroupingSize,
+
+  output: {  // required (can be an array, for multiple outputs)
+    // core output options
+    format,  // required
+    file,
+    dir,
+    name,
+    globals,
+
+    // advanced output options
+    paths,
+    banner,
+    footer,
+    intro,
+    outro,
+    sourcemap,
+    sourcemapFile,
+    sourcemapPathTransform,
+    interop,
+    extend,
+
+    // danger zone
+    exports,
+    amd,
+    indent,
+    strict,
+    freeze,
+    namespaceToStringTag,
+
+    // experimental
+    entryFileNames,
+    chunkFileNames,
+    assetFileNames
+  },
+
+  watch: {
+    chokidar,
+    include,
+    exclude,
+    clearScreen
+  }
+};
+```
+
+You can export an **array** from your config file to build bundles from several different unrelated inputs at once, even in watch mode. To build different bundles with the same input, you supply an array of output options for each input:
+
+```javascript
+// rollup.config.js (building more than one bundle)
+
+export default [{
+  input: 'main-a.js',
+  output: {
+    file: 'dist/bundle-a.js',
+    format: 'cjs'
+  }
+}, {
+  input: 'main-b.js',
+  output: [
+    {
+      file: 'dist/bundle-b1.js',
+      format: 'cjs'
+    },
+    {
+      file: 'dist/bundle-b2.js',
+      format: 'esm'
+    }
+  ]
+}];
+```
+
+If you want to create your config asynchronously, Rollup can also handle a `Promise` which resolves to an object or an array.
+
+```javascript
+// rollup.config.js
+import fetch from  'node-fetch';
+export default fetch('/some-remote-service-or-file-which-returns-actual-config');
```

Comment:
I think there might be a small mistake with these promise examples using fetch.
`fetch()` returns a promise that resolves to a `Response` which represents a status and a set of HTTP headers, etc. It does contain the response body and does not resolve to a plain object. In this example, you probably need to do something like:
```js
export default fetch('/some-remote-service-or-file-which-returns-actual-config').then(response => response.json());
```

Here is the [Reference](https://www.npmjs.com/package/node-fetch#json).